--- a/src/gallium/frontends/rusticl/api/device.rs
+++ b/src/gallium/frontends/rusticl/api/device.rs
@@ -299,7 +299,7 @@
             CL_DEVICE_UUID_KHR => cl_prop::<[cl_uchar; CL_UUID_SIZE_KHR as usize]>(
                 dev.screen().device_uuid().unwrap_or_default(),
             ),
-            CL_DEVICE_VENDOR => cl_prop::<&CStr>(dev.screen().device_vendor()),
+            CL_DEVICE_VENDOR => cl_prop::<&CStr>(if dev.screen().device_vendor().to_bytes() == b"AMD" { c"Advanced Micro Devices" } else { dev.screen().device_vendor() }),
             CL_DEVICE_VENDOR_ID => cl_prop::<cl_uint>(dev.vendor_id()),
             CL_DEVICE_VERSION => cl_prop::<&str>(&format!("OpenCL {} ", dev.cl_version.api_str())),
             CL_DRIVER_UUID_KHR => cl_prop::<[cl_char; CL_UUID_SIZE_KHR as usize]>(
